enrichments/trace/internal/elastic/span.go

```diff
@@ -520,6 +553,15 @@ func (s *spanEnrichmentContext) setInferredSpans(span ptrace.Span) {
 	}
 }
 
+func (s *spanEnrichmentContext) setUserAgentIfRequired(span ptrace.Span) {
+	if s.userAgentName == "" && s.inferredUserAgentName != "" {
+		span.Attributes().PutStr(semconv27.AttributeUserAgentName, s.inferredUserAgentName)
+	}
+	if s.userAgentVersion == "" && s.inferredUserAgentVersion != "" {
+		span.Attributes().PutStr(semconv27.AttributeUserAgentVersion, s.inferredUserAgentVersion)
+	}
```

Comment:
[**For reviewers**] I am setting the `user_agent.{name, version}` only if it is not already set.